ibm/service/cos/data_source_ibm_cos_backup_policy.go

```diff
@@ -0,0 +1,82 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package cos
+
+import (
+	"fmt"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	rc "github.com/IBM/ibm-cos-sdk-go-config/v2/resourceconfigurationv1"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceIBMCosBackupPolicy() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMCosBackupPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"bucket_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the source bucket",
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Id of the backup policy applied on the source bucket",
+			},
+			"policy_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the backup policy to be applied on the source bucket.",
+			},
+			"target_backup_vault_crn": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The CRN for a COS BackupVault.",
+			},
+			"backup_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of backup to support.",
+			},
+		},
+	}
+}
+
+func dataSourceIBMCosBackupPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
```

Comment:
As I remember to intialize CosConfigV1API client we need to construct url
something like this
```	satconfig := fmt.Sprintf("https://config.%s.%s.cloud-object-storage.appdomain.cloud/v1", serviceID, bLocation)
```
How are we initializing client just with bucket name?

Reply:
To create a client from CosCOnfugV1API we need the following -
`ibmcloud_api_key or bluemix_api_key or iam_token and iam_refresh_token`
We do not use the bucket name any where.